src/odemis/acq/move.py

```diff
@@ -243,8 +244,14 @@ def _doCryoSwitchSamplePosition(future, stage, focus, align, target):
 
             # Add the sub moves to perform the loading move
             sub_moves.append((stage, filter_dict({'rx', 'rz'}, stage_deactive)))
-            sub_moves.append((stage, filter_dict({'x', 'y'}, stage_deactive)))
-            sub_moves.append((stage, filter_dict({'z'}, stage_deactive)))
+            if current_label is UNKNOWN and not stage_referenced:
+                # After referencing the stage could move near the maximum axes range,
+                # and moving single axes may result in an invalid/reachable position error,
+                # so all axes will moved together for this special case.
+                sub_moves.append((stage, filter_dict({'x', 'y', 'z'}, stage_deactive)))
```

Comment:
Should this also be done in lines 271-272, if the stage has just come from LOADING (so it has run_reference())?

Reply:
If the stage is in LOADING position, then it's already referenced, so moving from there step by step should be fine.